### DISCUSSION TITLE
Numeric values in DAG details are incorrectly rendered as timestamps

### DISCUSSION THREAD

**Discussion by Gollum999** [apache/airflow#35430] [General] [answered]
### Apache Airflow version

2.7.2

### What happened

On the "Details" tab on a DAGs "Grid" page, all numeric DAG attributes are rendered as timestamps instead of numbers.  For example:

![image](https://github.com/apache/airflow/assets/7269927/9a99ccab-2d20-4a57-9fa8-63447e14444b)

I have noticed this behavior with the following fields, though there may be more:
* Max active runs
* Max active tasks
* Concurrency

### What you think should happen instead

Numeric fields should be rendered as numbers.

### How to reproduce

Go to any DAG's Grid page.  Don't select a DAG Run or Task Instance.  Click the Details tab if it is not already selected.

### Operating System

CentOS Stream 8

### Versions of Apache Airflow Providers

N/A

### Deployment

Other

### Deployment details

Standalone + self-hosted

### Anything else

I think the bug is [here](https://github.com/apache/airflow/blob/0a257afd031289062c76e7b77678337e88e10b93/airflow/www/static/js/dag/details/Dag.tsx#L133):
```
  // parse value for each key if date or not
  const parseStringData = (value: string) =>
    Number.isNaN(Date.parse(value)) ? value : <Time dateTime={value} />;
```
`Date.parse(1)` returns a number that is not NaN.

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by jscheffl** [comment]
Can you tell us a bit more about your setup and browser? I did a regression on 2.7.2 by using `docker run -it -p 38080:8080  apache/airflow:2.7.2-python3.8 bash` and in there called `airflow db reset` and then `AIRFLOW__CORE__LOAD_EXAMPLES=True airflow standalone`, started DAG `example_python_operator` and took a look to the GRID view. In my (Firefox) the numbers are shown correct:
![image](https://github.com/apache/airflow/assets/95105677/cf2cb6fe-44ce-4d7d-8e8d-70054f411d27)

Also tested with Cromium browser.

I need to admit, that the tested DAG does not define specific values but uses the fallback defaults. Is your case with a specific configuration in `max active runs` or `max active tasks`?

UPDATE: Also tested on latest main and was setting `max_active_runs=5` and `5`was correctly displayed in grid.

**Comment by potiuk** [comment]
You have forgotten to provide details about the DB you use @Gollum999 - but my guess is that your DB is causing the problem you see. probably it's something that is not exactly what Airflow standard requirements are.

**Reply by Gollum999** [reply]
Apologies, I am using Postgres 12.11.  I'm also using Chrome version 119.0.6045.105.

**Comment by potiuk** [comment]
Converting into discussion until this is clarified and mayb more details provided.

**Comment by Taragolis** [comment]
Seems like it not affects specific numbers and browsers. E.g. if number between [13.. 31] then everything work fine, however if it lower or greater than it replace by datetime value in Chromium, Chrome, Safari (I assume in all WebKit based) but works fine in Firefox. 
Some simple DAGs for test it it would be nice if someone else test it until we could reopen this issue, for avoid that OS specific stuff, some extensions side effects and etc.

I test it within:
- breeze + Postgres on macOS
- Airflow 2.7.3rc1 + MySQL on Linux

```python
import pendulum

from airflow.models.dag import DAG
from airflow.operators.empty import EmptyOperator

num_to_dt = {
    1: True,
    8: True,
    12: True,
    # 13-31 shows fine
    **{ix: False for ix in range(13, 32)},
    32: True,
    64: True,
    128: True,
    256: True,
}

for num, convert_to_dt in num_to_dt.items():
    with DAG(
        f"issue_35430_number_{num:03d}",
        start_date=pendulum.datetime(2023, 6, 1, tz="UTC"),
        schedule=None,
        catchup=False,
        max_active_runs=num,
        max_active_tasks=num,
        tags=["issue", "35430", "ui", f"int to dt: {convert_to_dt}", str(num)]
    ):
        EmptyOperator(task_id="empty", retries=num)
```

![image](https://github.com/apache/airflow/assets/3998685/9782d266-f894-4274-aa09-ec19224ab380)

![image](https://github.com/apache/airflow/assets/3998685/f3316ac5-c5ad-48c8-9df9-a99ac0c3f009)

![image](https://github.com/apache/airflow/assets/3998685/e5460a8b-7b31-4498-bf18-9c25b1f23ba7)

![image](https://github.com/apache/airflow/assets/3998685/2c0db9e0-baae-486c-93c0-3994a923484c)

![image](https://github.com/apache/airflow/assets/3998685/d3825fb6-e658-498f-80a6-7d6dc9ed140c)

**Comment by jscheffl** [comment, marked as answer]
Fixed with #35538
